recipes/compute_slurm_config.rb

```diff
@@ -35,10 +35,14 @@
   retry_delay 6
 end
 
-# Check to see if there is GPU on the instance, only execute run_nvidiasmi if there is GPU
+# Check to see if there is GPU on the instance, only execute run_nvidiasmi if there is GPU and nvidia installed
 if graphic_instance?
-  execute "run_nvidiasmi" do
-    command 'nvidia-smi'
+  if nvidia_installed?
+    execute "run_nvidiasmi" do
+      command 'nvidia-smi'
+    end
+  else
+    Chef::Log.warn("GPU instance but no Nvidia drivers found")
   end
 end
```

Comment:
we could remove this and have the nvidia_installed function execute nvidia-smi and return based on the exit code of the command